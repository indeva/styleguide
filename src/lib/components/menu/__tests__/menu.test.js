import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Menu from '..';

describe('Menu', () => {
  const defaultProps = {
    links: [
      { title: 'Visão Geral', icon: ['fas', 'store'] },
      {
        title: 'Relatórios',
        icon: ['fas', 'analytics'],
        active: true,
        links: [
          { title: 'Raio X' },
          { title: 'Eventos e motivos', active: true },
        ],
      },
    ],
  };

  it('should render properly', () => {
    const component = renderer.create(<Menu {...defaultProps} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render minified properly', () => {
    const component = renderer.create(<Menu {...defaultProps} />);
    component.root.instance.setState({ minified: true });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should minify on click on minified toggle', () => {
    const wrapper = mount(<Menu {...defaultProps} />);
    wrapper.find('.menu__header').simulate('click');
    expect(wrapper.find('.menu').hasClass('menu--minified')).toBeTruthy();
  });

  it('should back minification on second click on minified toggle', () => {
    const wrapper = mount(<Menu {...defaultProps} />);
    const menuHeader = wrapper.find('.menu__header');
    menuHeader.simulate('click');
    menuHeader.simulate('click');
    expect(wrapper.find('.menu').hasClass('menu--minified')).toBeFalsy();
  });
});
