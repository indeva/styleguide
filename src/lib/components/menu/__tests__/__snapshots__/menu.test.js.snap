// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should render minified properly 1`] = `
Array [
  <div
    className="class menu menu--expanded"
  >
    <div
      className="menu__header"
      onClick={[Function]}
      role="presentation"
    >
      <div
        className="menu__logo-wrapper"
      >
        <img
          className="menu__logo topbar__logo--in"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          src="logo-in.svg"
        />
        <img
          className="menu__logo topbar__logo--deva"
          src="logo-deva.svg"
        />
      </div>
      <img
        alt="Minificar menu"
        className="menu__burguer"
        src="burguer-icon.svg"
      />
    </div>
    <ul
      className="menu__nav"
    >
      <li
        className="menu__item"
      >
        <span
          className="menu__link"
          onClick={[Function]}
          role="presentation"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-store fa-w-20 menu__icon"
            data-icon="store"
            data-prefix="far"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 616 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M602 118.6L537.1 15C531.3 5.7 521 0 510 0H106C95 0 84.7 5.7 78.9 15L14 118.6c-29.6 47.2-10 110.6 38 130.8v227.4c0 19.4 14.3 35.2 32 35.2h448c17.7 0 32-15.8 32-35.2V249.4c48-20.2 67.6-83.6 38-130.8zM516 464H100v-96h416zm-.2-144.2H100v-64.7c24-3.3 45.1-15.2 60.3-32.2 18 20.1 44.3 33.1 73.8 33.1 29.6 0 55.8-13 73.8-33.1 18 20.1 44.3 33.1 73.8 33.1 29.6 0 55.8-13 73.8-33.1 15.3 17 36.3 28.9 60.3 32.2zm47.9-133c-3.2 6.8-10.9 18.6-27 20.8-2.4.3-4.8.5-7.2.5-14.7 0-28.2-6.1-38.1-17.2L455.7 151 420 190.8c-9.9 11.1-23.5 17.2-38.1 17.2s-28.2-6.1-38.1-17.2L308 151l-35.7 39.8c-9.9 11.1-23.5 17.2-38.1 17.2-14.7 0-28.2-6.1-38.1-17.2L160.3 151l-35.7 39.8c-9.9 11.1-23.5 17.2-38.1 17.2-2.5 0-4.9-.2-7.2-.5-16-2.2-23.8-13.9-27-20.8-5-10.8-7.1-27.6 2.3-42.6L114.8 48h386.3l60.2 96.1c9.5 15.1 7.5 31.9 2.4 42.7z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <span
            className="menu__item-text"
          >
            Vis찾o Geral
          </span>
        </span>
      </li>
      <li
        className="menu__item"
      >
        <span
          className="menu__link menu__link--active"
          onClick={[Function]}
          role="presentation"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-analytics fa-w-19 menu__icon menu__icon--active"
            data-icon="analytics"
            data-prefix="far"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 608 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 320h-64c-17.67 0-32 14.33-32 32v128c0 17.67 14.33 32 32 32h64c17.67 0 32-14.33 32-32V352c0-17.67-14.33-32-32-32zm-16 144h-32v-96h32v96zm176-272h-64c-17.67 0-32 14.33-32 32v256c0 17.67 14.33 32 32 32h64c17.67 0 32-14.33 32-32V224c0-17.67-14.33-32-32-32zm-16 272h-32V240h32v224zM256 192h-64c-17.67 0-32 14.33-32 32v256c0 17.67 14.33 32 32 32h64c17.67 0 32-14.33 32-32V224c0-17.67-14.33-32-32-32zm-16 272h-32V240h32v224zM96 352H32c-17.67 0-32 14.33-32 32v96c0 17.67 14.33 32 32 32h64c17.67 0 32-14.33 32-32v-96c0-17.67-14.33-32-32-32zM80 464H48v-64h32v64zM64 256c26.51 0 48-21.49 48-48 0-4.27-.74-8.34-1.78-12.28l101.5-101.5C215.66 95.26 219.73 96 224 96c6.15 0 11.97-1.26 17.38-3.37l95.34 76.27c-.35 2.33-.71 4.67-.71 7.1 0 26.51 21.49 48 48 48s48-21.49 48-48c0-2.43-.37-4.76-.71-7.09l95.34-76.27C532.03 94.74 537.85 96 544 96c26.51 0 48-21.49 48-48S570.51 0 544 0s-48 21.49-48 48c0 2.43.37 4.76.71 7.09l-95.34 76.27c-5.4-2.11-11.23-3.37-17.38-3.37s-11.97 1.26-17.38 3.37L271.29 55.1c.35-2.33.71-4.67.71-7.1 0-26.51-21.49-48-48-48s-48 21.49-48 48c0 4.27.74 8.34 1.78 12.28l-101.5 101.5C72.34 160.74 68.27 160 64 160c-26.51 0-48 21.49-48 48s21.49 48 48 48z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <span
            className="menu__item-text"
          >
            Relat처rios
          </span>
        </span>
        <ul
          className="menu__dropdown menu__dropdown--active"
        >
          <li
            className="menu__item menu__item--child"
          >
            <span
              className="menu__link menu__link--child"
              onClick={null}
              role="presentation"
            >
              Raio X
            </span>
          </li>
          <li
            className="menu__item menu__item--child"
          >
            <span
              className="menu__link menu__link--child menu__link--active"
              onClick={null}
              role="presentation"
            >
              Eventos e motivos
            </span>
          </li>
        </ul>
      </li>
    </ul>
  </div>,
  <div
    className="fullscreen-overlay menu__fullscreen_overlay menu__fullscreen_overlay--visible"
    onClick={[Function]}
  />,
]
`;

exports[`Menu should render properly 1`] = `
Array [
  <div
    className="class menu menu--expanded"
  >
    <div
      className="menu__header"
      onClick={[Function]}
      role="presentation"
    >
      <div
        className="menu__logo-wrapper"
      >
        <img
          className="menu__logo topbar__logo--in"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          src="logo-in.svg"
        />
        <img
          className="menu__logo topbar__logo--deva"
          src="logo-deva.svg"
        />
      </div>
      <img
        alt="Minificar menu"
        className="menu__burguer"
        src="burguer-icon.svg"
      />
    </div>
    <ul
      className="menu__nav"
    >
      <li
        className="menu__item"
      >
        <span
          className="menu__link"
          onClick={[Function]}
          role="presentation"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-store fa-w-20 menu__icon"
            data-icon="store"
            data-prefix="far"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 616 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M602 118.6L537.1 15C531.3 5.7 521 0 510 0H106C95 0 84.7 5.7 78.9 15L14 118.6c-29.6 47.2-10 110.6 38 130.8v227.4c0 19.4 14.3 35.2 32 35.2h448c17.7 0 32-15.8 32-35.2V249.4c48-20.2 67.6-83.6 38-130.8zM516 464H100v-96h416zm-.2-144.2H100v-64.7c24-3.3 45.1-15.2 60.3-32.2 18 20.1 44.3 33.1 73.8 33.1 29.6 0 55.8-13 73.8-33.1 18 20.1 44.3 33.1 73.8 33.1 29.6 0 55.8-13 73.8-33.1 15.3 17 36.3 28.9 60.3 32.2zm47.9-133c-3.2 6.8-10.9 18.6-27 20.8-2.4.3-4.8.5-7.2.5-14.7 0-28.2-6.1-38.1-17.2L455.7 151 420 190.8c-9.9 11.1-23.5 17.2-38.1 17.2s-28.2-6.1-38.1-17.2L308 151l-35.7 39.8c-9.9 11.1-23.5 17.2-38.1 17.2-14.7 0-28.2-6.1-38.1-17.2L160.3 151l-35.7 39.8c-9.9 11.1-23.5 17.2-38.1 17.2-2.5 0-4.9-.2-7.2-.5-16-2.2-23.8-13.9-27-20.8-5-10.8-7.1-27.6 2.3-42.6L114.8 48h386.3l60.2 96.1c9.5 15.1 7.5 31.9 2.4 42.7z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <span
            className="menu__item-text"
          >
            Vis찾o Geral
          </span>
        </span>
      </li>
      <li
        className="menu__item"
      >
        <span
          className="menu__link menu__link--active"
          onClick={[Function]}
          role="presentation"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-analytics fa-w-19 menu__icon menu__icon--active"
            data-icon="analytics"
            data-prefix="far"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 608 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 320h-64c-17.67 0-32 14.33-32 32v128c0 17.67 14.33 32 32 32h64c17.67 0 32-14.33 32-32V352c0-17.67-14.33-32-32-32zm-16 144h-32v-96h32v96zm176-272h-64c-17.67 0-32 14.33-32 32v256c0 17.67 14.33 32 32 32h64c17.67 0 32-14.33 32-32V224c0-17.67-14.33-32-32-32zm-16 272h-32V240h32v224zM256 192h-64c-17.67 0-32 14.33-32 32v256c0 17.67 14.33 32 32 32h64c17.67 0 32-14.33 32-32V224c0-17.67-14.33-32-32-32zm-16 272h-32V240h32v224zM96 352H32c-17.67 0-32 14.33-32 32v96c0 17.67 14.33 32 32 32h64c17.67 0 32-14.33 32-32v-96c0-17.67-14.33-32-32-32zM80 464H48v-64h32v64zM64 256c26.51 0 48-21.49 48-48 0-4.27-.74-8.34-1.78-12.28l101.5-101.5C215.66 95.26 219.73 96 224 96c6.15 0 11.97-1.26 17.38-3.37l95.34 76.27c-.35 2.33-.71 4.67-.71 7.1 0 26.51 21.49 48 48 48s48-21.49 48-48c0-2.43-.37-4.76-.71-7.09l95.34-76.27C532.03 94.74 537.85 96 544 96c26.51 0 48-21.49 48-48S570.51 0 544 0s-48 21.49-48 48c0 2.43.37 4.76.71 7.09l-95.34 76.27c-5.4-2.11-11.23-3.37-17.38-3.37s-11.97 1.26-17.38 3.37L271.29 55.1c.35-2.33.71-4.67.71-7.1 0-26.51-21.49-48-48-48s-48 21.49-48 48c0 4.27.74 8.34 1.78 12.28l-101.5 101.5C72.34 160.74 68.27 160 64 160c-26.51 0-48 21.49-48 48s21.49 48 48 48z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <span
            className="menu__item-text"
          >
            Relat처rios
          </span>
        </span>
        <ul
          className="menu__dropdown menu__dropdown--active"
        >
          <li
            className="menu__item menu__item--child"
          >
            <span
              className="menu__link menu__link--child"
              onClick={null}
              role="presentation"
            >
              Raio X
            </span>
          </li>
          <li
            className="menu__item menu__item--child"
          >
            <span
              className="menu__link menu__link--child menu__link--active"
              onClick={null}
              role="presentation"
            >
              Eventos e motivos
            </span>
          </li>
        </ul>
      </li>
    </ul>
  </div>,
  <div
    className="fullscreen-overlay menu__fullscreen_overlay menu__fullscreen_overlay--visible"
    onClick={[Function]}
  />,
]
`;
